make einsum unbacked friendly

ColinPeppler opened the pull request:
cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @chenyang78 @kadeng @muchulee8 @amjames @chauhang @aakhundov

ColinPeppler committed:
make einsum unbacked friendly

[ghstack-poisoned]

ColinPeppler committed:
Update on "make einsum unbacked friendly"




cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy chenyang78 kadeng muchulee8 amjames chauhang aakhundov

[ghstack-poisoned]

ColinPeppler committed:
Update on "make einsum unbacked friendly"




cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy chenyang78 kadeng muchulee8 amjames chauhang aakhundov

[ghstack-poisoned]

ColinPeppler commented:
@pytorchbot rebase

ColinPeppler committed:
Update

[ghstack-poisoned]

ColinPeppler commented:
@pytorchbot rebase

ColinPeppler committed:
Update on "make einsum unbacked friendly"




cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy chenyang78 kadeng muchulee8 amjames chauhang aakhundov

[ghstack-poisoned]

pianpwk commented on test/inductor/test_unbacked_symints.py:
```diff
@@ -442,6 +442,29 @@ def forward(self, arg0_1, arg1_1, arg2_1, arg3_1, arg4_1, arg5_1, arg6_1):
+            qk = torch.einsum("bcxd,bcyd->bcxy", (q, k))
+            qk2 = torch.einsum("b...,b...->b...", (q, k))
+            qvec = torch.einsum("b...,b->b...", (q, vector))
+            return qk, qk2, qvec
```
q: do you not have to mark the symbol size-like to make this work? Is it done inside the repeat call maybe?

ColinPeppler replied:
yup exactly done inside repeat call

ColinPeppler committed:
Update on "make einsum unbacked friendly"




cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy chenyang78 kadeng muchulee8 amjames chauhang aakhundov

Differential Revision: [D72874052](https://our.internmc.facebook.com/intern/diff/D72874052)

[ghstack-poisoned]

ColinPeppler commented:
@pytorchbot merge

ColinPeppler commented:
@pytorchbot merge

ColinPeppler committed:
Update on "make einsum unbacked friendly"

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy chenyang78 kadeng muchulee8 amjames chauhang aakhundov

[ghstack-poisoned]
